app/models/assignment.rb

```diff
@@ -1156,6 +1156,31 @@ def to_json(options = {})
     self.assignment_properties.attributes.merge(self.attributes).symbolize_keys.to_json(options)
   end
 
+  # Returns an assignment's relevant properties for uploading/downloading an assignment's configuration as a hash
+  def assignment_properties_config
+    properties = {}
+    # Insert DEFAULT_FIELDS first to make file more familiar
+    DEFAULT_FIELDS.each do |f|
+      properties[f] = self.assignment.send(f)
+    end
+    # Get all other assignment properties
+    other_properties = self.assignment_properties.attributes.merge(self.attributes).symbolize_keys
+    section_dates = SectionDueDate.where(assessment_id: self.id).pluck_to_hash(:due_date, :section_id, :start_time)
+    required_files = AssignmentFile.where(assessment_id: self.id).pluck_to_hash(:filename)
+    late_policy = SubmissionRule.where(assessment_id: self.id)
+                                .joins('submission_rules LEFT JOIN periods
+                                        ON periods.submission_rule_id = submission_rules.id')
+                                .pluck_to_hash(:type, :submission_rule_id, 'periods.id',
```

Comment:
I considered that but I am concerned that some information may be lost/duplicated. Namely, the schema seems to suggest that an assignment can have multiple submission types. If I were to remove the ids, I'm concerned that I would not know which period belongs to which submission_rule. 
I tried to make the hash hierarchical to fix this but the code I ended up writing was pretty inefficient compared to just joining the two tables.  
Any suggestions on how to approach this would be appreciated.

Reply:
Never mind, I was looking at the schema.rb file for a UNIQUE keyword but after taking a look at the top of the assignment.rb file, it seems there is only one submission_rule per assignment. That makes a lot more sense. I will definitely make this change then